src/simple_graph.py

```diff
@@ -41,26 +42,31 @@ def del_node(self, node):
             del self.graph_dict[node]
             for key in self.graph_dict:
                 if node in self.graph_dict[key]:
-                    self.graph_dict[key].remove(node)
+                    del self.graph_dict[key][node]
         except KeyError:
             raise KeyError('Node does not exist')
 
     def del_edge(self, node1, node2):
         """Delete the edge connecting node 1 and node 2."""
         try:
-            self.graph_dict[node1].remove(node2)
+            self.graph_dict[node1]
         except KeyError:
             raise KeyError('Node1 does not exist')
-        except ValueError:
+        try:
+            del self.graph_dict[node1][node2]
+        except KeyError:
             raise ValueError('Node2 does not exist')
 
     def has_node(self, node):
         """Return a boolean for if node exists."""
-        return isinstance(self.graph_dict.get(node), list)
+        return isinstance(self.graph_dict.get(node), dict)
 
     def neighbors(self, node):
         """Return the list of all nodes connected to by edges."""
-        return self.graph_dict.get(node)
+        try:
+            return set(self.graph_dict.get(node).keys())
+        except AttributeError:
+            return None
```

Comment:
I can get why you'd return a None, but since the node type is dict it might make more sense to return an empty dict.